File: ios/DetonatorCamera/Sources/DetonatorCamera/CameraRequest/CameraStartRecordingRequest.swift
import Detonator

class CameraStartRecordingRequest: Request {
    override func run() -> Void {
        let edge = getComponentEdge()
        
        let element = edge!.children[0].element as! CameraElement
        
        element.startRecording() { result in
            switch result {
            case .success(let media):
                self.end(data: media)
            case .failure(let error):
                self.error(error: error)
            }
        }
    }
}

File: android/DetonatorCamera/src/main/java/com/iconshot/detonator/camera/camerarequest/CameraStartRecordingRequest.java
package com.iconshot.detonator.camera.camerarequest;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.camera.CameraMedia;
import com.iconshot.detonator.renderer.Edge;
import com.iconshot.detonator.request.Request;

import com.iconshot.detonator.camera.cameraelement.CameraElement;

public class CameraStartRecordingRequest extends Request {
    public CameraStartRecordingRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        Edge edge = getComponentEdge();

        CameraElement element = (CameraElement) edge.children.get(0).element;

        element.startRecording(new CameraElement.CaptureCallback() {
            public void onEnd(CameraMedia media) {
                end(media);
            }

            public void onError(Exception exception) {
                error(exception);
            }
        });
    }
}
